torch/serialization.py

```diff
@@ -453,14 +453,20 @@ def persistent_load(saved_id):
             raise RuntimeError("Unknown saved id type: %s" % saved_id[0])
 
     f_is_real_file = _is_real_file(f)
-    if f_is_real_file and f.tell() == 0:
-        # legacy_load requires that f has fileno()
-        # only if offset is zero we can attempt the legacy tar file loader
-        try:
-            return legacy_load(f)
-        except tarfile.TarError:
-            # if not a tarfile, reset file offset and proceed
-            f.seek(0)
+
+    try:
+        if f_is_real_file and f.tell() == 0:
+            # legacy_load requires that f has fileno()
+            # only if offset is zero we can attempt the legacy tar file loader
+            try:
+                return legacy_load(f)
+            except tarfile.TarError:
+                # if not a tarfile, reset file offset and proceed
+                f.seek(0)
+    except (io.UnsupportedOperation, AttributeError) as e:
+        msg = (str(e) + ". You can only torch.load from a file that is seekable. " +
+               "Please pre-load the data into a buffer like io.BytesIO and try to load from it instead.")
+        raise type(e)(msg)
```

Comment:
Getting an `AttributeError` or `UnsupportedOperation` in here doesn't imply that it has been caused by seek in any way. This check has to be improved, or the error message will be highly misleading.

Reply:
Ah, I see. Would it be helpful if I show this error message only when specific errors, such as "seek", are caught?